src/lib/scheduler_clients/slurm/slurm_rest_client.py

```diff
@@ -89,6 +89,12 @@ async def submit_job(
         username: str,
         jwt_token: str,
     ) -> int | None:
+
+        # if `script_path`` is set and `script` is not provided,
+        # then this cannot be submitted with SLURM API
+        if job_description.script_path and not job_description.script:
```

Comment:
If we aim to abstract the scheduler type and its behaviour, should we transparently handle this case?
Instead of returning an error, we could force the use of the  slurm_cli_client.py
To do so, change the scheduler client dependency to: `Depends(SchedulerClientDependency(force_cli_client=True)),` on the [post job route](https://github.com/eth-cscs/firecrest-v2/blob/48985b91f736a07900cde5a0b4dfd241280e8135/src/firecrest/compute/router.py#L46C11-L46C26)

The flip side is that job submission via Slurm REST API will never be called. At least until Slurm implements the ability to provide a script path also via REST API.

Reply:
I was thinking about that idea. 
My main concern here is that this is only possible with SLURM. What if the scheduler used as API doesn't provide a CLI version installed in the same system?

Reply:
Agreed, the current implementation is not ideal. 
With `force_cli_client=True`, it is the route function that implements the logic of using the CLI or the REST API. Instead, this decision should be delegated to the SchedulerClientDependency class or the Slurm clint class.

I would proceed as follows:

- for the time being, use: `Depends(SchedulerClientDependency(force_cli_client=True))`
- open an issue to remove the `force_cli_client` parameter. Instead, we should create a new Slurm client that integrates both the CLI and the REST clients and "automatically" routes the requests via the correct channel.